modules/WordCloudPlugin/src/main/java/net/clementlevallois/wordcloud/InitialWordProcessingRunnable.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * author: Clément Levallois
+ */
+package net.clementlevallois.wordcloud;
+
+import java.util.ResourceBundle;
+import org.gephi.graph.api.GraphModel;
+import org.gephi.utils.longtask.spi.LongTask;
+import org.gephi.utils.progress.Progress;
+import org.gephi.utils.progress.ProgressTicket;
+import org.openide.util.NbBundle;
+
+public class InitialWordProcessingRunnable implements LongTask, Runnable {
+
+    private ProgressTicket progressTicket;
+    private boolean cancelled = false;
+    private final GraphModel graphModel;
+    private final String columnName;
+    private final String lang;
+    private static final ResourceBundle bundle = NbBundle.getBundle(LexplorerTopComponent.class);
+
+    public InitialWordProcessingRunnable(GraphModel graphModel, String columnName, String lang) {
+        this.graphModel = graphModel;
+        this.columnName = columnName;
+        this.lang = lang;
+    }
+
+    @Override
+    public void run() {
+        Progress.setDisplayName(progressTicket, bundle.getString("progress.initial_analysis_running"));
+        Progress.start(progressTicket);
+        TopTermExtractor topTermExtractor = new TopTermExtractor();
+        topTermExtractor.tokenizeSelectedTextualAttributeForTheEntireGraph(graphModel, columnName, lang);
+        Progress.finish(progressTicket, bundle.getString("progress.initial_analysis_complete"));
+
+    }
+
+    @Override
+    public boolean cancel() {
+        this.cancelled = true;
+        Progress.finish(progressTicket, bundle.getString("progress.initial_analysis_interrupted"));
```

Comment:
In your current configuration, this runnable won't be interrupted. If you want the `LongTaskExecutor` to kill your Thread when `cancel` is called, you need to initialise it with a `interruptDelay`, for instance 
```new LongTaskExecutor(true, "Executor", 1000)```. In this example, the thread will be killed 1 second after cancel has been hit. If no interrupt delay is passed, the executor doesn't interrupt so cancel has essentially no effect.

That all said, it seems like a very bad idea to interrupt the thread (ie. a dirty cancel) as you are iterating over the nodes. While you iterate over the nodes you hold a read lock on the graph. If the thread is interrupted, the read lock won't be released and Gephi will eventually hang. 

I would advise to either not make this task cancellable and deal with it, or refactor so that the `tokenizeSelectedTextualAttributeForTheEntireGraph` method can be cancelled.

Reply:
I have added a Boolean flag in the while loop which iterates over the nodes. This flag is turned to 'false' when the user clicks on the cancel Button, before excecutor.cancel() is called.